feat: ZoomContentControl as a stand alone control in Toolkit (#1249)
unoplatform/uno.toolkit.ui: doc/controls/ZoomContentControl.md

## Summary

`ZoomContentControl` allows you to display content that can be zoomed in and out, as well as panned. It is especially useful for scenarios such as viewing large images, maps, or documents where users need control over zoom levels and panning.

### C\#

```csharp
public partial class ZoomContentControl : Control
```

### XAML

```xml
xmlns:utu="using:Uno.Toolkit.UI"
...

<utu:ZoomContentControl ZoomLevel="1.5"
                        MinZoomLevel="0.5"
                        MaxZoomLevel="3.0"
                        IsZoomAllowed="True"
                        IsPanAllowed="True">
    <utu:ZoomContentControl.Content>
        <Image Source="ms-appx:///Assets/Media/LargeMedia.svg"
               Stretch="Uniform" />
    </utu:ZoomContentControl.Content>
</utu:ZoomContentControl>
```

### Inheritance

`Object` &#8594; `DependencyObject` &#8594; `UIElement` &#8594; `FrameworkElement` &#8594; `Control` &#8594; `ContentControl` &#8594; `ZoomContentControl`

### Constructors

| Constructor| Description|
|----------------|-------------------------------------------------------|
| `ZoomContentControl()`|Initializes a new instance of the `ZoomContentControl` class.|

> ### Properties

Reviewer: beside certain private properties for intermediary values, you have some properties are not listed here:
- AutoZoomToCanvasOnSizeChanged
- IsActive

> | Property| Type| Description|
> |------------------------|-----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
> | `ZoomLevel`| `double`| Gets or sets the current zoom level for the content.|

Reviewer: we discussed about the formatting, you should either format to align the table, or
just use single - for header separator and left single space on each side of cell content:
```suggestion
| Property| Type | Description |
|-|-|-|
| `ZoomLevel`| `double` | Gets or sets the current zoom level for the content. |
```

| `MinZoomLevel`| `double`| Gets or sets the minimum zoom level allowed for the content.|
| `MaxZoomLevel`| `double`| Gets or sets the maximum zoom level allowed for the content.|

> | `isZoomAllowed`| `bool`| Gets or sets a value indicating whether zooming is allowed.|

Reviewer 2: Capitalization isZoom => IsZoom...

> | Property               | Type                  | Description                                                                                                                                                                                                                                                                                                                                            |
> |------------------------|-----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
> | `ZoomLevel`          | `double`         | Gets or sets the current zoom level for the content.                                                                                                                                                                                                                                                                                             |
> | `MinZoomLevel`              | `double`              | Gets or sets the minimum zoom level allowed for the content.                                                                                                                                                                                                                   |

Reviewer: if you are not manually adjusting the column width or some formatting tools,
just ignore the align, and pad with single space like so:
```suggestion
| Property | Type | Description |
|-|-|-|
| `ZoomLevel` | `double` | Gets or sets the current zoom level for the content. |
| `MinZoomLevel` | `double` | Gets or sets the minimum zoom level allowed for the content. |
```

> | `isZoomAllowed` | `bool` | Gets or sets the value indicating whether zooming is allowed.                                                                                                                                             |
> | `IsPanAllowed` | `bool` | Gets or sets the value indicating whether panning is allowed.                                                        |

Reviewer: keep the same wording, as the other lines and same as microsoft's
```suggestion
| `isZoomAllowed` | `bool` | Gets or sets a value indicating whether zooming is allowed.                                                                                                                                             |
| `IsPanAllowed` | `bool` | Gets or sets a value indicating whether panning is allowed.                                                        |
```

| `HorizontalZoomCenter`| `double`| Gets or sets the horizontal center point of the zooming operation.|
| `VerticalZoomCenter`| `double`| Gets or sets the vertical center point of the zooming operation.|
| `ScaleWheelRatio`| `double`| Gets or sets the ratio for scaling zoom level when using a mouse wheel.|
| `PanWheelRatio`| `double`| Gets or sets the ratio for panning when using a mouse wheel.|
| `ResetWhenNotActive`| `bool`| Gets or sets a value indicating whether the zoom and pan should reset when the control is not active.|

### Methods

| Method| Return Type| Description|
|-----------------------|------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `ResetZoom()`| `void`| Resets the zoom level to its default value.|
| `ResetOffset()`| `void`| Resets the horizontal and vertical offset to its default value. |

> | `ZoomToCanvas()`| `void`| Adjust the zoom level so that the content fits within the available space.|
> | `Centralize()`| `void`| Centers the content within the available space.|

Reviewer: please rename these methods:
```suggestion
| `FitToCanvas()`| `void`| Adjust the zoom level so that the content fits within the available space.|
| `CenterContent()`| `void`| Centers the content within the available space.|
```

| `VerticalZoomCenter` | `double` | Gets or sets the vertical center point of the zooming operation. |
| `ScaleWheelRatio` | `double` | Gets or sets the ratio for scaling zoom level when using a mouse wheel. |
| `PanWheelRatio` | `double` | Gets or sets the ratio for panning when using a mouse wheel. |
| `ResetWhenNotActive` | `bool` | Gets or sets a value indicating whether the zoom and pan should reset when the control is not active. |
| `AutoZoomToCanvasOnSizeChanged` | `bool` | Gets or sets a value indicating whether the control should automatically adjust zoom when resized. |
| `IsActive` | `bool` | Gets or sets a value indicating whether the control is active. |

### Methods

| Method| Return Type| Description|
|-----------------------|------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `ResetZoom()`| `void`| Resets the zoom level to its default value.|

Reviewer 3: No longer public

> | `ResetOffset()`| `void`| Resets the horizontal and vertical offset to its default value. |

Reviewer 3: No longer public
